pkg/sync/syncer.go

```diff
@@ -1856,6 +1856,10 @@ func (s *syncer) processGrantsWithExternalPrincipals(ctx context.Context, princi
 
 	for _, grant := range grants {
 		annos := annotations.Annotations(grant.Annotations)
+		if !annos.Contains(&v2.ExternalResourceMatchAll{}) && !annos.Contains(&v2.ExternalResourceMatch{}) &&
```

Comment:
It would be nice if we had annos.ContainsAny() which would help us avoid iterating over the annotations three times here.